service/src/main/java/bio/terra/cbas/dao/RunSetDao.java

```diff
@@ -23,23 +23,39 @@ public RunSetDao(NamedParameterJdbcTemplate jdbcTemplate) {
     this.jdbcTemplate = jdbcTemplate;
   }
 
-  public List<RunSet> getRunSets() {
+  public List<RunSet> getRunSets(Integer pageSize) {
     String sql =
         "SELECT * FROM run_set "
             + "INNER JOIN method_version ON run_set.method_version_id = method_version.method_version_id "
-            + "INNER JOIN method on method_version.method_id = method.method_id GROUP BY run_set.run_set_id, method_version.method_version_id, method.method_id ORDER BY MIN(run_set.submission_timestamp) DESC";
-    return jdbcTemplate.query(sql, new RunSetMapper());
+            + "INNER JOIN method on method_version.method_id = method.method_id GROUP BY run_set.run_set_id, method_version.method_version_id, method.method_id ORDER BY MIN(run_set.submission_timestamp) DESC LIMIT :pageSize";
+    return jdbcTemplate.query(
+        sql, new MapSqlParameterSource("pageSize", pageSize), new RunSetMapper());
   }
 
-  public RunSet getRunSet(UUID methodId) {
+  public RunSet getRunSet(UUID runSetId) {
     String sql =
         "SELECT * FROM run_set "
             + "INNER JOIN method_version ON run_set.method_version_id = method_version.method_version_id "
             + "INNER JOIN method on method_version.method_id = method.method_id "
-            + "WHERE method.method_id = :methodId GROUP BY run_set.run_set_id, method_version.method_version_id, method.method_id "
+            + "WHERE run_set.run_set_id = :runSetId GROUP BY run_set.run_set_id, method_version.method_version_id, method.method_id "
             + "ORDER BY MIN(run_set.submission_timestamp) DESC";
     return jdbcTemplate
-        .query(sql, new MapSqlParameterSource("methodId", methodId), new RunSetMapper())
+        .query(sql, new MapSqlParameterSource("runSetId", runSetId), new RunSetMapper())
+        .get(0);
+  }
+
+  public RunSet getRunSetWithMethodId(UUID methodId, Integer pageSize) {
+    String sql =
+        "SELECT * FROM run_set "
+            + "INNER JOIN method_version ON run_set.method_version_id = method_version.method_version_id "
+            + "INNER JOIN method on method_version.method_id = method.method_id "
+            + "WHERE method.method_id = :methodId GROUP BY run_set.run_set_id, method_version.method_version_id, method.method_id "
```

Comment:
This `GROUP BY` is a little surprising to me, since I would think we would want the entire results set rather than some grouped version

Reply:
I don't remember the exact details, but when trying to use `ORDER BY` I was getting an error on `run_set.run_set_id` about needing there to be an `ORDER BY` clause. After adding it, it gave the same error for the other two columns. Adding them solved the error. Apologies I can't remember exactly what is was.